### PR TITLE
Resolve the decorator type as type and check if the symbol has value.

### LINKED ISSUE #4239: Require been ignord for emitDecoratorMetadata if module is unused and name is lowercase
If an imported module is unused it will be ignored, which is good (It may exist edge cases where this is bad but that is an other discussion).

```javascript
import {db} from './db'
console.log('Hello Dexter Morgan');
```
Compiles to:
```javascript
console.log('Hello Dexter Morgan');
```
(One can of course, `import './db'` to keep it).

But this becomes a problem if I use `emitDecoratorMetadata` and a decorator:

```javascript
import {db} from './db';

function someDecorator(target) {
    return target;
}

@someDecorator
class MyClass {
    db: db;

    constructor(db: db) {
        this.db = db;
        this.db.doSomething();
    }
}

export {MyClass};
```

This compiles to:
```javascript
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") return Reflect.decorate(decorators, target, key, desc);
    switch (arguments.length) {
        case 2: return decorators.reduceRight(function(o, d) { return (d && d(o)) || o; }, target);
        case 3: return decorators.reduceRight(function(o, d) { return (d && d(target, key)), void 0; }, void 0);
        case 4: return decorators.reduceRight(function(o, d) { return (d && d(target, key, o)) || o; }, desc);
    }
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
function someDecorator(target) {
    return target;
}
var MyClass = (function () {
    function MyClass(db) {
        this.db = db;
        this.db.doSomething();
    }
    MyClass = __decorate([
        someDecorator, 
        __metadata('design:paramtypes', [db_1.db]) // db_1 is not defined
    ], MyClass);
    return MyClass;
})();
exports.MyClass = MyClass;
```

And if a run this code I get `ReferenceError: db_1 is not defined`.

The funny thing is however that this code works:

```javascript
import {db as Database} from './db';

function someDecorator(target) {
    return target;
}

@someDecorator
class MyClass {
    db: Database;

    constructor(db: Database) {
        this.db = db;
        this.db.doSomething();
    }

}

export {MyClass}
```

If I import the database class with `import {Database} from './db';` it works as well. So it seams like the compiler ignore to import a module if its name is in lower case? 

I'm using version `1.6.0-dev.20150808` of `tsc`.

### PR DESCRIPTION
**Pull request by sheetalkamat**
This would make sure we are referencing correct symbol to check if it has value
Fixes #4239

### CODE REVIEW

File: src/compiler/checker.ts [hunk 1]
```diff
@@ -11384,9 +11384,16 @@ namespace ts {
             // serialize the type metadata.
             if (node && node.kind === SyntaxKind.TypeReference) {
                 let root = getFirstIdentifier((<TypeReferenceNode>node).typeName);
-                let rootSymbol = resolveName(root, root.text, SymbolFlags.Value, /*nameNotFoundMessage*/ undefined, /*nameArg*/ undefined);
-                if (rootSymbol && rootSymbol.flags & SymbolFlags.Alias && !isInTypeQuery(node) && !isConstEnumOrConstEnumOnlyModule(resolveAlias(rootSymbol))) {
-                    markAliasSymbolAsReferenced(rootSymbol);
+                var meaning = root.parent.kind === SyntaxKind.TypeReference ? SymbolFlags.Type : SymbolFlags.Namespace;
```
Comment (mhegazy) [inline src/compiler/checker.ts]: not sure i understand this line.

Comment (sheetalkamat) [inline src/compiler/checker.ts]: If it is M.Q then resolving M has to be like namespace. While if it is T , T need sto be resolved as type.

Comment (yuit) [inline src/compiler/checker.ts]: use ```let```

**Comment by yuit** [comment]
Please add more tests

**Comment by mhegazy** [comment]
:+1: but please add some more tests for the namespace/dotted name case. this area does not have enough coverage.
also i would add a unit test in https://github.com/Microsoft/TypeScript/blob/master/tests/cases/unittests/transpile.ts to cover this example in ts.transpile cases.

**Comment by mhegazy** [comment]
:+1:

**Pull request merged**